explorer/ui/app/components/queries/block.tsx

```diff
@@ -15,42 +15,48 @@ export function GetBlocksInRange(from: number, to: number): XBlock[] {
   var rows: XBlock[] = []
 
   data?.xblockrange.map((xblock: any) => {
-    var msgs: XMsg[] = []
     let block = {
       id: xblock.BlockHeight,
       UUID: '',
       SourceChainID: xblock.SourceChainID,
       BlockHash: xblock.BlockHash,
       BlockHeight: xblock.BlockHeight,
-      Messages: msgs,
+      Messages: [],
       Timestamp: xblock.Timestamp,
       Receipts: [],
     }
 
-    xblock.Messages.map((msg: any) => {
-      let xmsg = {
-        DestAddress: '',
-        DestChainID: '',
-        DestGasLimit: '',
-        SourceChainID: '',
-        SourceMessageSender: '',
-        StreamOffset: '',
-        TxHash: '',
-        BlockHeight: '',
-        BlockHash: '',
-        Receipts: [],
-        Block: block,
-      }
-      msgs.push(xmsg)
-    })
-
-    block.Messages = msgs
     rows.push(block)
   })
 
   return rows
 }
 
+export function GetBlock(sourceChainID: number, height: number): XBlock {
+  const [result] = useQuery({
+    query: xblock,
+    variables: {
+      sourceChainID: '0x' + sourceChainID.toString(16),
+      height: '0x' + height.toString(16),
+    },
+  })
+  const { data, fetching, error } = result
+
+  if (data?.xblock === undefined || data?.xblock === null) {
+    return {
+      BlockHeight: height,
+      Messages: [],
+      Receipts: [],
+      BlockHash: '',
+      SourceChainID: '',
+      Timestamp: '',
+      UUID: '',
+    }
+  }
+
+  return data?.xblock as XBlock
```

Comment:
if we are using these in components, let's return the full `useQuery` result. It has a bunch of useful stuff. Also - we can generate these use query hooks w/ proper types with our codegen tool

Second - let's try not to do these queries in components, and instead do them in [loaders](https://remix.run/docs/en/main/route/loader). It seems that just using them in components does not get us the SSR we want

Reply:
Wrote these helpers for Morne, just trying to make it easier and generate the code neccessary for him to call in the loaders

Reply:
You can't use react hooks in the loader func,  so these won't work for that. You'll need to just query the client directly.

Reply:
just do a `client.query` instead, or something like that

Reply:
yeah talked with him this morning, going to pull these out helpers out rn

Reply:
@kevinhalliday should be all set now